backend/common/src/main/java/ai/verta/modeldb/common/futures/Future.java

```diff
@@ -380,4 +400,12 @@ public InternalFuture<T> toInternalFuture() {
   static void resetExecutorForTest() {
     futureExecutor = null;
   }
+
+  public static void enableCallSiteStackCapture() {
+    System.setProperty(FUTURE_TESTING_STACKS_ENABLED, "true");
+  }
+
+  public static void disableCallSiteStackCapture() {
+    System.setProperty(FUTURE_TESTING_STACKS_ENABLED, "false");
+  }
```

Comment:
Yeah, definitely a static field.  I think it should default to the value of `DEEP_TRACING_ENABLED`, but can be overridden.  Maybe have one method `public static void forceCallSiteStackCapture(boolean value)`?  If `DEEP_TRACING_ENABLED` is `true`, this method has no effect.

Reply:
so, are you suggesting that we don't allow enabling this via system property at JVM startup, but only programmatically?

Reply:
ok, best of both worlds is probably to initialize a static field based on the system property, but then allow overriding via these methods. I don't think we should do any special handling around the DEEP_TRACING_ENABLED value, since they only coincidentally happen to share the stack trace capture.